source/FreeRTOS_IP.c

```diff
@@ -271,10 +276,31 @@ static void prvProcessIPEventsAndTimers( void )
     /* Wait until there is something to do. If the following call exits
      * due to a time out rather than a message being received, set a
      * 'NoEvent' value. */
-    if( xQueueReceive( xNetworkEventQueue, ( void * ) &xReceivedEvent, xNextIPSleep ) == pdFALSE )
-    {
-        xReceivedEvent.eEventType = eNoEvent;
-    }
+    #if ipconfigEVENT_QUEUES > 1
+        if( ulTaskNotifyTake( pdFALSE, xNextIPSleep ) == pdFALSE )
```

Comment:
Makes sense to decrement it by making the first parameter 'pdFALSE'. Behaves like a counting semaphore.

Reply:
I don't understand your comment. From https://www.freertos.org/ulTaskNotifyTake.html . The first parameter is the clear on exit and the second parameter the time to wait. Isn't the call exactly like it should be? Wait for the notification value and decrement it on exit. Only look at the event queue if the notification value is not 0.

Reply:
I was confused, sorry. :D